src/main/scala/org/constellation/concurrency/cuckoo/CuckooFilter.scala

```diff
@@ -0,0 +1,25 @@
+package org.constellation.concurrency.cuckoo
+
+import java.io.ByteArrayInputStream
+
+import net.cinnom.nanocuckoo.NanoCuckooFilter
+import org.constellation.schema.snapshot.FilterData
+
+case class CuckooFilter(filterData: FilterData) {
+
+  private lazy val filter: NanoCuckooFilter = {
+    val stream = new ByteArrayInputStream(filterData.contents)
+    try {
+      val resultingFilter = new NanoCuckooFilter.Builder(filterData.capacity).build()
+      resultingFilter.readMemory(stream)
```

Comment:
I am wondering if we should wrap it by  `F`. I mean creating stream, reading into stream and closing stream can throw 🤔

Reply:
Good question. I will try to make it inside of F.

Reply:
@kpudlik this is just reading from memory, there's no IO here and no side effects. Actually, I think it should never throw. I added `try .. finally` just for safety. Do you still think that I should try to make it inside of `F`?